transaction/core/src/membership_proofs/mod.rs

```diff
@@ -30,7 +29,7 @@ lazy_static! {
 pub fn hash_leaf(tx_out: &TxOut) -> [u8; 32] {
     let mut hasher = Blake2b256::new();
     hasher.update(&TXOUT_MERKLE_LEAF_DOMAIN_TAG);
-    tx_out.digest(&mut hasher);
+    hasher.update(&tx_out.hash());
```

Comment:
i didn't want to mess around with this for now, we try to make this use merlin instead of blake2b though
i think its fine either way